test/services/sync/strategies/webuntis/WebUntisSyncer.test.js

```diff
@@ -0,0 +1,197 @@
+		});
+	});
+
+	it('runs with an inclusive list of metadataIds', async () => {
+		const { _id: datasourceId } = await testObjects.createTestDatasource({
+			config: { target: 'webuntis' },
+		});
+		await Promise.all(new Array(20).fill('').map(
+			() => app.service('webuntisMetadata').create({
+				datasourceId,
+				teacher: 'Renz',
+				class: '2a',
+				room: '0-23',
+				subject: 'mathe',
+				state: 'new',
+			}),
+		));
+		const metadata = await app.service('webuntisMetadata')
+			.find({ query: { datasourceId, $limit: 5 } });
+		const toImport = metadata.data.map((md) => md._id);
+		const args = WebUntisSyncer.params({
+			datasourceId,
+			query: {
+				username: 'webuntisusername',
+				password: 'secret123',
+				url: 'url.com',
+			},
+		}, {
+			datatype: 'inclusive',
+			courseMetadataIds: toImport,
+		});
+		const syncer = new WebUntisSyncer(app, {}, undefined, ...args);
+		const result = await syncer.sync();
+		expect(result.success).to.eq(true);
+		const updatedMetadata = await app.service('webuntisMetadata').find({ query: { datasourceId } });
+		expect(updatedMetadata.total).to.be.greaterThan(0);
+		updatedMetadata.data.forEach((el) => {
+			if (toImport.includes(el._id)) {
+				expect(el.state).to.equal('updated');
+			} else {
+				expect(el.state).to.equal('discarded');
+			}
+		});
+	});
+
+	it('runs with an exclusive list of metadataIds', async () => {
+		const { _id: datasourceId } = await testObjects.createTestDatasource({
+			config: { target: 'webuntis' },
+		});
+		await Promise.all(new Array(20).fill('').map(
+			() => app.service('webuntisMetadata').create({
+				datasourceId,
+				teacher: 'Renz',
+				class: '2a',
+				room: '0-23',
+				subject: 'mathe',
+				state: 'new',
+			}),
+		));
+		const metadata = await app.service('webuntisMetadata')
+			.find({ query: { datasourceId, $limit: 5 } });
+		const toDiscard = metadata.data.map((md) => md._id);
+		const args = WebUntisSyncer.params({
+			datasourceId,
+			query: {
+				username: 'webuntisusername',
+				password: 'secret123',
+				url: 'url.com',
+			},
+		}, {
+			datatype: 'exclusive',
+			courseMetadataIds: toDiscard,
+		});
+		const syncer = new WebUntisSyncer(app, {}, undefined, ...args);
+		const result = await syncer.sync();
+		expect(result.success).to.eq(true);
+		const updatedMetadata = await app.service('webuntisMetadata').find({ query: { datasourceId } });
+		expect(updatedMetadata.total).to.be.greaterThan(0);
+		updatedMetadata.data.forEach((el) => {
+			if (toDiscard.includes(el._id)) {
+				expect(el.state).to.equal('discarded');
+			} else {
+				expect(el.state).to.equal('imported');
+			}
+		});
+	});
+
+	it('may be triggered via datasources', async () => {
+		const { _id: datasourceId } = await testObjects.createTestDatasource({
+			config: {
+				target: 'webuntis',
+				username: 'untisUsername',
+				password: 'secret123',
+				url: 'webuntis.org',
+			},
+		});
+		const { _id: runId } = await app.service('datasourceRuns').create({
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['runId' is assigned a value but never used.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4899819)